chore/check-javadoc-regressions.py

```diff
@@ -0,0 +1,272 @@
+    prints all identified regressions.
+    """
+    lines_reference = filter_relevant_lines(reference.splitlines())
+    lines_other = filter_relevant_lines(other.splitlines())
+
+    new_errors = find_added_violation_lines(
+        Counter(map(strip_line_number, lines_reference)),
+        Counter(map(strip_line_number, lines_other))
+    )
+    with_line_numbers = try_readd_line_numbers(new_errors, lines_other)
+
+    print(warn("You likely added the following new errors. "), end=" ")
+    print("Line numbers might be incorrect if there are multiple violations of the same type in the file.")
+    for error in with_line_numbers:
+        print(error.strip())
+
+
+def print_compare_with_branch(target_branch: str) -> None:
+    """
+    Compares the current branch with a given reference branch and prints a change summary.
+    Exits with an error if the current branch has more violations than the passed target branch.
+    """
+    # Switch to the root so checkstyle will run against the whole project
+    os.chdir(run_command(["git", "rev-parse", "--show-toplevel"]).strip())
+
+    with temporary_path(suffix=".xml") as config_path:
+        write_checkstyle_config(config_path)
+
+        run_command(["git", "checkout", target_branch], stderr=PIPE)
+        reference_output = run_checkstyle(config_path)
+        run_command(["git", "checkout", "-"], stderr=PIPE)
+        other_output = run_checkstyle(config_path)
+
+        reference_violation_count = extract_violation_count(reference_output)
+        other_violation_count = extract_violation_count(other_output)
+
+        if reference_violation_count is None:
+            print(warn("The Checkstyle run for the reference branch did not yield any result."))
+            exit(1)
+        if other_violation_count is None:
+            print(warn("The Checkstyle run for your branch did not yield any result."))
+            exit(1)
+
+        print()
+        print("Current status")
+        print(
+            f"  Violations on {hl('comparison')} branch ({target_branch}): "
+            f"{hl(str(reference_violation_count))}")
+        print(
+            f"  Violations on {hl('your')} branch         {' '*len(target_branch)}: "
+            f"{hl(str(other_violation_count))}"
+        )
+        print()
+
+        if reference_violation_count < other_violation_count:
+            print(warn("Javadoc quality has deteriorated!"))
+            print("Run the chore/ci-checkstyle-javadoc.sh script locally to find errors")
+            print(f"See {hl('https://github.com/inria/spoon/issues/3923')} for details")
+            print()
+            print_regression_lines(reference_output, other_output)
+            exit(1)
+        elif reference_violation_count == other_violation_count:
+            print(success("Javadoc quality has not deteriorated!"))
+        else:
+            print(success("You increased the Javadoc quality! Thank you :)"))
```

Comment:
This is a very minor note, but it seems potentially confusing to me to say that one has "increased" the quality, when that corresponds to the quality score decreasing (recall that the score is just the amount of warnings).

```suggestion
            print(success("You improved the Javadoc quality! Thank you :)"))
```